### ISSUE TITLE
The "outputs" interestingness test is not working in Python 3+

### ISSUE THREAD

**Issue by nth10sd** [MozillaSecurity/lithium#33]
I installed lithium in a virtualenv for Python 3 on master (without any of our imports patches) and found that the "outputs" interestingness test is not working as expected:

```
$ touch temp.js
$ testVenv/bin/python -m lithium outputs --timeout=2 temp.js ls temp.js
Intermediate files will be stored in tmp11/.
The original testcase has 0 lines.
Checking that the original testcase is 'interesting'...
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/__main__.py", line 13, in <module>
    main()
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/lithium.py", line 1322, in main
    exit(Lithium().main())
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/lithium.py", line 1105, in main
    return self.run()
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/lithium.py", line 1121, in run
    result = self.strategy.main(self.testcase, self.interesting, self.testcaseTempFilename)
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/lithium.py", line 248, in main
    if not interesting(testcase, writeIt=False):
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/lithium.py", line 1276, in interesting
    inter = self.conditionScript.interesting(self.conditionArgs, tempPrefix)
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/../interestingness/outputs.py", line 39, in interesting
    fileIngredients.fileContains(tempPrefix + "-out.txt", searchFor, regexEnabled)[0] or
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/../interestingness/fileIngredients.py", line 16, in fileContains
    return fileContainsStr(f, s, verbose=vb), s
  File "/home/gkwubu/aa20170728/testVenv/lib/python3.5/site-packages/lithium/lithium/../interestingness/fileIngredients.py", line 23, in fileContainsStr
    if line.find(s) != -1:
TypeError: a bytes-like object is required, not 'str'
```

**Comment by nth10sd** [comment]
Perhaps related to the with_statements [here](https://github.com/MozillaSecurity/lithium/blob/master/interestingness/fileIngredients.py#L20) and [here](https://github.com/MozillaSecurity/lithium/blob/master/interestingness/fileIngredients.py#L36) in fileIngredients.py opening in `rb` mode.

Should we, for the first instance, change [line 21](https://github.com/MozillaSecurity/lithium/blob/master/interestingness/fileIngredients.py#L21) to:
```
gContents = [l.decode("utf8", "replace").strip() for l in g.read().splitlines() if l.strip()]
```
and the second at [line 37](https://github.com/MozillaSecurity/lithium/blob/master/interestingness/fileIngredients.py#L37) to:
```
foundRegex = regex.search(g.read().decode("utf8", "replace"))
```
Both of the following tests fail prior to the changes above, and succeed after:
```
$ testVenv/bin/python -m lithium outputs --timeout=2 temp.js ls temp.js
```
and
```
$ testVenv/bin/python -m lithium outputs --timeout=2 --regex '^temp' ls temp.js
```

**Comment by nth10sd** [comment]
This is a [sample fix](https://github.com/MozillaSecurity/lithium/pull/28/commits/735fbc74479bda34d6071b211b068b6b383c680a) (committed temporarily to a PR, may be overwritten and disappear later).

**Comment by jschwartzentruber** [comment]
We shouldn't assume UTF-8 encoding.

See: [fileIngredients.py](https://github.com/MozillaSecurity/lithium/compare/master...jschwartzentruber:total_recall#diff-49d65856f0a71df7adb0942369146b62) (UTF-8 used for print only, not for matching) and [outputs.py](https://github.com/jschwartzentruber/lithium/blob/total_recall/interestingness/outputs.py#L41-L42) for an alternate fix.

**Comment by jschwartzentruber** [comment]
Fixed in #39
